app/helpers/PermissionHints.php

```diff
@@ -0,0 +1,36 @@
+<?php
+namespace App\Helpers;
+
+use Nette\Reflection\ClassType;
+use Nette\Reflection\Method;
+use Nette\StaticClassException;
+use Nette\Utils\Strings;
+
+class PermissionHints {
+  public function __construct() {
+    throw new StaticClassException();
+  }
+
+  public static function generate($aclModule, $subject) {
+    $reflection = ClassType::from($aclModule);
+    foreach ($reflection->getMethods(Method::IS_PUBLIC) as $method) {
+      if (!Strings::startsWith($method->getName(), "can")) {
+        continue;
+      }
+
+      if ($method->getNumberOfRequiredParameters() > 1 || $method->getNumberOfParameters() === 0) {
+        continue;
+      }
+
+      $parameter = $method->getParameters()[0];
+      $className = $parameter->getClass() ? $parameter->getClass()->getName() : null;
+      if ($className !== null && $subject instanceof $className) {
+        yield lcfirst(substr($method->getName(), 3)) => $method->invoke($aclModule, $subject);
+      }
+    }
+  }
```

Comment:
I must say I am not happy about this solution. I do not mind using reflection, but imagine when the same type of entity (e.g., exercise) is decorated in the view using this hints generator many times. It is rather crude to iterate over an ACL class every time hints need to be added to an entity. I would suggest introducing some caching or adjusting the interface so that a generator (which already knows the right ACL methods) is constructed for the view, so the view may use it more efficiently to add hints to multiple entities.